sematic/resolvers/cloud_resolver.py

```diff
@@ -245,7 +245,34 @@ def _detach_resolution(self, future: AbstractFuture) -> str:
         return run.id
 
     def _schedule_future(self, future: AbstractFuture) -> None:
-        run = api_client.schedule_run(future.id)
+        run = api_client.get_run(future.id)
+        if run.future_state == FutureState.SCHEDULED.value:
+            # It's unclear how we wind up in this situation, but it shouldn't be fatal.
+            # Log some info to help diagnose, and move on.
+            logger.warning(
+                "Tried to double schedule %s. Futures: %s. "
+                "Runs: %s. Buffer runs: %s. Retrieved run: %s",
+                run.id,
+                self._futures,
+                self._runs,
+                self._buffer_runs,
+                run,
+            )
+        else:
+            try:
+                run = api_client.schedule_run(future.id)
+                logger.info("Scheduled run: %s", run)
+            except Exception:
+                logger.error(
+                    "Error scheduling run %s. Futures: %s. "
+                    "Runs: %s. Buffer runs: %s. Retrieved run: %s",
+                    run.id,
+                    self._futures,
+                    self._runs,
+                    self._buffer_runs,
+                    run,
+                )
+                raise
         self._runs[run.id] = run
```

Comment:
Maybe use `self._add_run` here?

Reply:
done

Reply:
Actually, this caused problems. Mentioned them in a comment